### ISSUE TITLE
SSL connection fails with build 6858 and later

### ISSUE THREAD

**Issue by evs-xsarus** [HeidiSQL/HeidiSQL#2034]
### Description

Connection to a MySQL 8.0 instance running on a docker, I have to enable 'Use SSL' for the connection. On a new installation, this didn't work anymore.

I was able to pinpoint it to build 6858. The previous build, 6857, was the last build that could connect to my db-server.

### HeidiSQL version

6858

### Database server version

MySQL 8.0.39 docker

### Reproduction recipe

Try to connect to the server.

### Error/Backtrace

```shell
After a few seconds, HeidiSQL shows a 'connection lost during query'-error.
```

**Comment by ansgarbecker** [comment]
6858 is 5b046b2473b633c206c825739cb3f92bddb571b3

The deprecated `mysql_ssl_set` was not used any longer, instead just several calls to `mysql_options`.

I would be happy to help out here, but I have no clue what I should try out.

Did you already try the different libraries, including the newer libmysql-8.4.dll?

**Comment by ansgarbecker** [comment]
You said you "have to enable 'Use SSL' for the connection". Does that mean you leave the other SSL options empty? Perhaps you could show your session settings per screenshot.

**Comment by evs-xsarus** [comment]
Yes, all other options are left empty. Ticking the box feels like HeidiSQL will poll the server in a way that the server accepts it. I've also tested it with build 6908 using the installer. Same result, tried all dlls.

Settings:
![image](https://github.com/user-attachments/assets/d44da227-06f6-4a5b-9584-6869ff95e2be)

SSH tunnel:
![image](https://github.com/user-attachments/assets/a5b45880-54da-41cc-976e-5942890caddd)

Advanced:
![image](https://github.com/user-attachments/assets/f7da25b3-0f20-4c96-aa98-b6a134fae5bf)

SSL:
![image](https://github.com/user-attachments/assets/aa647e2f-9ae5-4452-8977-931ebe025a98)

**Comment by evs-xsarus** [comment]
PS Without settings, I can only use libmariadb.dll. The others give this error:

libmysql-6.1.dll
![image](https://github.com/user-attachments/assets/f738aacd-8c1b-49d1-94d5-a56324adaf3d)

libmysql-8.4.0.dll:
![image](https://github.com/user-attachments/assets/843c2cbb-847d-4eef-b625-def3e876dcf3)

libmysql.dll:
![image](https://github.com/user-attachments/assets/ba56bd23-8336-426b-bae5-a2481e3e8efb)

**Comment by ansgarbecker** [comment]
> PS Without settings, I can only use libmariadb.dll. The others give this error

Does that mean libmariadb.dll works normally?

**Comment by ansgarbecker** [comment]
libmysql-8.4 tells you what is happening: "certificate verify failed". Same what is documented in https://github.com/HeidiSQL/HeidiSQL/issues/1972#issuecomment-2196945544
Try setting the dropdown (in the latest release) to "No verification".
